mill-plugin/src/test/scala/mill/util/TestEvaluator.scala

```diff
@@ -0,0 +1,132 @@
+package mill.util
+
+import java.io.{InputStream, PrintStream}
+import mill.define.{Input, Target, Task}
+import mill.api.Result.OuterStack
+import mill.eval.Evaluator
+import mill.api.Strict.Agg
+import utest.assert
+import utest.framework.TestPath
+
+import language.experimental.macros
+import mill.api.{DummyInputStream, Result}
+
+// Borrowed from the mill codebase : https://github.com/Baccata/mill/blob/master/main/test/src/util/TestEvaluator.scala
+object TestEvaluator {
```

Comment:
I've resorted to copying this test evaluator from mill's codebase, as it allows to tests plugins in unit tests (from SBT, or any build tool), which makes it drastically easier to set up than having SBT shell-out to a mill process in order to run integration tests.

Reply:
Alternatively, is it possible to just run mill's integration tests as a separate stage? 
We no longer have sbt-gha so if it's easier, you can just add a mill stage to the workflow?

Reply:
> We no longer have sbt-gha so if it's easier, you can just add a mill stage to the workflow? 

The problem is really that you'd have to add an explicit call to `publishLocal` from SBT before being able to run the tests from mill, which is like, totally fair enough in CI, but utterly shite in terms of developer experience on local machines

Reply:
Alright.

Not fantastic that a copypasta has to live here, so I sure hope at some point it can be made into a dependency.
At this point it's more important to have a mill plugin + tests, so let's keep it this way.

Reply:
> so I sure hope at some point it can be made into a dependency. 

I hope so too, I have a number of usecases for this